### PR TITLE
Feature/main menu console fix viewers

### PR DESCRIPTION
**Pull request by MariuszMalec**
TmpDataBase.cs shift to folder Services, set minimal console screen, fixed TransactionViewer nad UserViewer, added Balance to viewers transactions and added readkey and comments before finish any options, fixed ViewTransactionEdit

### CODE REVIEW

File: TheBTeam.ConsoleApp/MainMenu.cs [hunk 1]
```diff
@@ -32,7 +32,9 @@ public static void ShowMainMenu()
                 {
                     Console.Clear();
                     Console.WriteLine("------------------------------------------");
-                    Console.WriteLine($"Welcome in the financial planner");
+                    Console.WriteLine($"Welcome in the financial planner         ");
```
Comment (PuchalskiMateusz) [inline TheBTeam.ConsoleApp/MainMenu.cs:35]: Welcome in the financial planner by TheBTeam
maybe will be more eye catching

File: TheBTeam.ConsoleApp/MainMenu.cs [hunk 2]
```diff
@@ -32,7 +32,9 @@ public static void ShowMainMenu()
...
+                    Console.WriteLine("------------------------------------------");
+                    Console.WriteLine("---------------TheBTeam-------------------");
```
Comment (PuchalskiMateusz) [inline TheBTeam.ConsoleApp/MainMenu.cs:37]: Get rid of it if above variation will be ok

File: TheBTeam.ConsoleApp/MainMenu.cs [hunk 3]
```diff
@@ -69,6 +71,8 @@ public static void ShowMainMenu()
                 {
                     Console.WriteLine($"{mainMenuItem[currentItem]} ...");
                     tmpListUsers.UsersList = LoadDataFromFile.ReadUserFile();
```
Comment (PuchalskiMateusz) [inline TheBTeam.ConsoleApp/MainMenu.cs:73]: There is no information that daata was loaded succesfully or not

File: TheBTeam.ConsoleApp/MainMenu.cs [hunk 4]
```diff
@@ -69,6 +71,8 @@ public static void ShowMainMenu()
...
+                    Console.WriteLine($"Press any key to continue");
```
Comment (PuchalskiMateusz) [inline TheBTeam.ConsoleApp/MainMenu.cs:74]: There is no information that daata was loaded succesfully or not

File: TheBTeam.ConsoleApp/MainMenu.cs [hunk 5]
```diff
@@ -81,8 +85,17 @@ public static void ShowMainMenu()
...
-                    UserViewer.ViewUsers(tmpListUsers.UsersList);
-                    Console.ReadKey();
+                    if (tmpListUsers.UsersList.Count > 0)
+                    {
+                        UserViewer.ViewUsers(tmpListUsers.UsersList);
+                        Console.ReadKey();
+                    }
+                    else
+                    {
+                        Console.WriteLine($"No data users, Please add new user");
```
Comment (PuchalskiMateusz) [inline TheBTeam.ConsoleApp/MainMenu.cs:95]: No data, please add new user or
no users pleas add new user

File: TheBTeam.ConsoleApp/MainMenu.cs [hunk 6]
```diff
@@ -97,20 +110,34 @@ public static void ShowMainMenu()
...
-                    TransactionViewer.ViewTransaction(tmpListTransactions.TransactionsList);
-                    Console.ReadKey();
+                    if (tmpListTransactions.TransactionsList.Count != 0)
+                    {
+                        TransactionViewer.ViewTransaction(tmpListTransactions.TransactionsList);
+                        Console.ReadKey();
+                    }
+                    else
+                    {
+                        Console.WriteLine($"No data transactions, Please enter transaction");
```
Comment (PuchalskiMateusz) [inline TheBTeam.ConsoleApp/MainMenu.cs:120]: no data or no transaction

File: TheBTeam.ConsoleApp/MainMenu.cs [hunk 7]
```diff
@@ -97,20 +110,34 @@ public static void ShowMainMenu()
...
-                    TransactionViewer.ViewTransactionAccordingCategory(tmpListTransactions.TransactionsList);
-                    Console.ReadKey();
+                    if (tmpListTransactions.TransactionsList.Count != 0)
+                    {
+                        TransactionViewer.ViewTransactionAccordingCategory(tmpListTransactions.TransactionsList);
+                        Console.ReadKey();
+                    }
+                    else
+                    {
+                        Console.WriteLine($"No data transactions, Please enter transaction");
```
Comment (PuchalskiMateusz) [inline TheBTeam.ConsoleApp/MainMenu.cs:135]: like above

File: TheBTeam.ConsoleApp/MainMenu.cs (hunk 5 above)
Comment (PuchalskiMateusz) [reply TheBTeam.ConsoleApp/MainMenu.cs:95]: maby add sth like
--------------------------------------------------------------------------------------
and later too

**Review by PuchalskiMateusz — CHANGES_REQUESTED** [review]

File: TheBTeam.ConsoleApp/MainMenu.cs [hunk 8]
```diff
@@ -97,20 +118,34 @@ public static void ShowMainMenu()
...
                 }
                 else if (mainMenuItem[currentItem] == ("Show transaction according Category"))
                 {
                     Console.WriteLine($"{mainMenuItem[currentItem]}");
-                    TransactionViewer.ViewTransactionAccordingCategory(tmpListTransactions.TransactionsList);
-                    Console.ReadKey();
+                    if (tmpListTransactions.TransactionsList.Count != 0)
+                    {
+                        TransactionViewer.ViewTransactionAccordingCategory(tmpListTransactions.TransactionsList);
+                        Console.ReadKey();
```
Comment (PuchalskiMateusz) [inline TheBTeam.ConsoleApp/MainMenu.cs:139]: move it to methods! guess it will be cleaner and more practical

Comment (PuchalskiMateusz) [reply TheBTeam.ConsoleApp/MainMenu.cs:139]: about end

File: TheBTeam.ConsoleApp/TransactionViewer.cs [hunk 9]
```diff
@@ -9,24 +9,30 @@ public static class TransactionViewer
...
                 Console.WriteLine($"|{item.User.FirstName.ToString().PadRight(textPaddingWidth, paddingChar)} " +
                                   $"|{item.User.LastName.ToString().PadRight(textPaddingWidth, paddingChar)} " +
                                   $"|{item.Type.ToString().PadRight(textPaddingWidth, paddingChar)}" +
                                   $"|{item.Category.ToString().PadRight(textPaddingWidth, paddingChar)}" +
                                   $"|{item.Currency.ToString().PadRight(textPaddingWidth, paddingChar)}" +
+                                  $"|{item.User.Balance.ToString().PadRight(textPaddingWidth, paddingChar)}" +
                                   $"|{item.Amount.ToString().PadRight(textPaddingWidth, paddingChar)}");
             }
+            Console.WriteLine(("").PadRight(textPaddingWidth * numberOfCollumn, '='));
+            Console.WriteLine($"Press any key to continue");
```
Comment (PuchalskiMateusz) [inline TheBTeam.ConsoleApp/TransactionViewer.cs:35]: here you can add readkey

File: TheBTeam.ConsoleApp/TransactionViewer.cs [hunk 10]
```diff
@@ -9,24 +9,30 @@ public static class TransactionViewer
     {
         public static void ViewTransaction(List<Transaction> transactions)
```
Comment (PuchalskiMateusz) [inline TheBTeam.ConsoleApp/TransactionViewer.cs:10]: What about null list?
Should me statem to check it and give information!

File: TheBTeam.ConsoleApp/TransactionViewer.cs [hunk 11]
```diff
@@ -35,34 +41,50 @@ public static void ViewTransactionAccordingCategory(List<Transaction> transactio
...
             var tmpTransactions = transactions.Where(t => t.User.Email == email).Where(t => t.Category == categoryOfTransaction);
-            var textPaddingWidth = 23;
+            var textPaddingWidth = 20;
             var paddingChar = ' ';
-            Console.WriteLine($"|{"Category".PadRight(textPaddingWidth, paddingChar)} " +
+            var numberOfCollumn = 5;
+
+            if (tmpTransactions.ToList().Count > 0)
+            {
+                Console.WriteLine(("").PadRight(textPaddingWidth * numberOfCollumn, '='));
```
Comment (PuchalskiMateusz) [inline TheBTeam.ConsoleApp/TransactionViewer.cs:50]: Why do you repeta this Code? you can use Method ViewTransaction(tmpTransactions)

File: TheBTeam.ConsoleApp/TransactionViewer.cs [hunk 12]
```diff
@@ -35,34 +41,50 @@ public static void ViewTransactionAccordingCategory(List<Transaction> transactio
             var categoryOfTransaction = ConsoleFactory.GetCategoryOfTransaction();
             Console.WriteLine($"{categoryOfTransaction}");
             var tmpTransactions = transactions.Where(t => t.User.Email == email).Where(t => t.Category == categoryOfTransaction);
```
Comment (PuchalskiMateusz) [inline TheBTeam.ConsoleApp/TransactionViewer.cs:43]: This is logic, i thing it should be in Bll.TransactionService

Comment (PuchalskiMateusz) [reply TheBTeam.ConsoleApp/TransactionViewer.cs:43]: what more i Jira, task was: Wyszukiwanie transakcji po kategorii
no info about user

**Review by PuchalskiMateusz — CHANGES_REQUESTED** [review]

File: TheBTeam.ConsoleApp/TransactionViewer.cs (hunk 9 above)
Comment (MariuszMalec) [reply TheBTeam.ConsoleApp/TransactionViewer.cs:35]: Herę readkey is not needed , is in main menu

File: TheBTeam.ConsoleApp/MainMenu.cs (hunk 8 above)
Comment (MariuszMalec) [reply TheBTeam.ConsoleApp/MainMenu.cs]: I shifted it to transaction viewer from main menu

**Pull request merged**